updating/updating-mongodb-documents-with-go.md

```diff
@@ -0,0 +1,84 @@
+
+- Go 1.13
+- MongoDB Go Driver 1.1.2
+- Visual Studio Code (VS Code)
+- MongoDB Atlas with an M0 cluster (FREE)
+
+While you may be able to use different versions than I'm using and find success, my recommendation is to try to match what I'm using as best as possible.
+
+> It is FREE to use an M0 cluster on Atlas, but if you'd like some premium credit applied to your account for a more powerful cluster, you can use promotional code NRABOY200 within Atlas.
+
+If you need help connecting your application to MongoDB Atlas, check out my tutorial [Quick Start: How to Get Connected to Your MongoDB Cluster with Go](https://), which goes through the basics.
+
+## Updating Data within a Collection
+
+If you've been keeping up with every tutorial in the series for getting started with Go, you'll remember that we're working with a `podcasts` collection and an `episodes` collection. As a quick refresher, Documents in the `podcasts` collection might look like this:
+
+```json
+{
+    "_id": ObjectId("5d9e0173c1305d2a54eb431a"),
+    "title": "The Polyglot Developer Podcast",
+    "author": "Nic Raboy"
+}
+```
+
+At some point in time you're going to find yourself needing to update Documents, whether they look like my example, or reflect something of your own design. With the Go Driver for MongoDB, there are numerous options towards updating Documents.
+
+When updating a Document, you'll need filter criteria as well as what should be updated within the data that matched the filter.
+
+Take the following for example:
+
+```golang
+podcastsCollection := client.Database("quickstart").Collection("podcasts")
+id, _ := primitive.ObjectIDFromHex("5d9e0173c1305d2a54eb431a")
+result, err := podcastsCollection.UpdateOne(
+    ctx,
+    bson.M{"_id": id},
+    bson.D{
+        {"$set", bson.D{{"author", "Nic Raboy"}}},
+    },
+)
+if err != nil {
+    log.Fatal(err)
+}
+fmt.Printf("Updated %v Documents!\n", result.MatchedCount)
+```
+
+In the above example, we connect to our `podcasts` collection, which at this point we're going to assume has some Documents in it by the previously mentioned design. If not, fill in the gap with your own collections or Documents.
+
+When using the `UpdateOne` function on a collection, we're choosing to update only the first Document that matches our filter, in this case an exact match on Document ID. If a match was found, the `author` field on that document will be set. The filter and change criteria can leverage the full scope of the MongoDB Query Language (MQL).
+
+The `UpdateResult`, as seen as the `result` variable, has metric information about the operation. For example, it might say how many Documents were updated.
+
+Let's say we wanted to update more than one Document. We could make use of the `UpdateMany` function on a collection. Take the following for example:
+
+```golang
+result, err = podcastsCollection.UpdateMany(
+    ctx,
+    bson.M{"title": "The Polyglot Developer Podcast"},
```

Comment:
I sat on this for a while to think it over.

My personal opinion is that it doesn't matter. There can be shows with the same name, but it is unlikely. If anything, the show URL would be unique, but we don't use it in the example.

If you are strongly against me leaving `podcast` as the data for this tutorial, let me know and I'll change it. Otherwise, let's keep it as is.

Reply:
I'd pretty strongly prefer to change this, given that there's the easy use case of updating every episode in a podcast. If you can't think of a good use case for updating every episode, though, it's probably fine to leave this.

Reply:
Given the data model that I presented for the `episodes` collection, there wouldn't be a use case for updating every episode. Remember, the schema looks like this:

```
{
    "_id": ObjectID
    "podcast": ObjectID
    "title": String,
    "duration": Number
}
```

In reality it is quite similar to the `podcasts` collection. I mean we could update all of the ObjectID for the show, but realistically that should never happen.

Reply:
Fair point. It's fine to leave this as-is then.